engine/src/rocksdb_engine.rs

```diff
@@ -386,13 +385,13 @@ impl RocksSnapshot {
     }
 
     /// Read data from the snapshot
-    async fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
-        if buf.is_empty() {
+    async fn read(&mut self, buf: &mut BytesMut) -> io::Result<usize> {
```

Comment:
The `read` method doesn't depend on any specific methods of `BytesMut`, so, IMO, we can make the second parameter more general. How about `impl BufMut`?

Reply:
The `read_buf_mut` implementation rely on `len` and `capacity` funtions, which are not trait methods. So I think it's better keep using `BytesMut`.